bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gems/CanopusWfs.java

```diff
@@ -70,14 +75,64 @@ public boolean asterismPreFilter(final ImList<SiderealTarget> targets) {
             return checkAsterismMagnitude(targets);
         }
 
-        public boolean checkAsterismMagnitude(final ImList<SiderealTarget> targets) {
+        // Ensure that targets are not more than MAGNITUDE_DIFFERENCE_LIMIT apart in magnitude in R.
+        public static boolean checkAsterismMagnitude(final ImList<SiderealTarget> targets) {
             final ImList<Double> sortedMags = extractRMagnitudes(targets);
             if (sortedMags.isEmpty() || sortedMags.size() != targets.size())
                 return false;
             return (sortedMags.size() == 1) || (sortedMags.last() - sortedMags.head() <= MAGNITUDE_DIFFERENCE_LIMIT);
         }
 
-        public ImList<Double> extractRMagnitudes(final ImList<SiderealTarget> targets) {
+        // Guide windows cannot overlap.
+        // As these are square, we need a position angle to determine if they overlap or not.
+        @Override
+        public boolean asterismFilter(final ObsContext ctx, final ImList<SiderealTarget> targets) {
+            return findOverlappingGuideWindows(ctx, targets).isEmpty();
+        }
+
+        // Given an ObsContext (we need a position angle) and a list of targets, find the list of pairs of targets
+        // whose guide windows overlap.
+        public static ImList<Pair<SiderealTarget, SiderealTarget>> findOverlappingGuideWindows(final ObsContext ctx, final ImList<SiderealTarget> targets) {
+            final double posAngle = ctx.getPositionAngle().toRadians();
+
+            // Check each pair of guide windows to make sure they do not overlap.
+            final List<Pair<SiderealTarget, SiderealTarget>> overlappingGuideWindows = new ArrayList<>(3);
+
+            // Must use indices here to avoid double pairs of the form (t1,t2) and (t2,t1).
+            for (int i=0; i < targets.size(); ++i) {
+                final SiderealTarget t1 = targets.get(i);
+                final Area a1 = windowFromCoordinates(posAngle, t1.coordinates());
+                for (int j=i+1; j < targets.size(); ++j) {
+                    final SiderealTarget t2 = targets.get(j);
+                    final Area a2 = windowFromCoordinates(posAngle, t2.coordinates());
+
+                    // To check overlap, take a1 xor a2 and compare to a1 union a2.
+                    // If they are not equal, they overlap.
```

Comment:
Would it not just be: `!a1.intersect(a2).isEmpty` implies overlap?

Reply:
Yes, that's much better (although `intersect` returns `void` and mutates the original object). 
I'll clone `a1`, intersect with `a2`, and then check `!intArea.isEmpty()`.

I'll make the change and test it. Thanks for helping me to simplify this.